### PR TITLE
Added addon label test as an integration test

### PR DESCRIPTION
**Pull request by Ankit152**
Signed-off-by: Ankit Kurmi <akurmi@redhat.com>

## Description
Added addon label test as an integration test

### CODE REVIEW

File: integration/mtcli/label_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("validate subcommand", func() {
+	type labelTestCase struct {
+		MetadataPath  string
+		ShouldSucceed bool
+	}
+
+	DescribeTable("validator subcommand",
```
Comment (Ajpantuso) [inline integration/mtcli/label_test.go:22]:
```suggestion
	DescribeTable("AM0002 enabled",
```

File: internal/testdata/metadata_v1/legacy/connectors-operator/metadata/stage/addon.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  required: true
+  value_type: string
+- description: Client secret for connecting to the SSO authentication server
+  editable: true
+  enabled: true
+  id: client-secret
+  name: SSO Client Secret
+  required: true
+  value_type: string
+addonOwner: Red Hat OpenShift Connectors team <mas-connectors@redhat.com>
```
Comment (Ajpantuso) [inline internal/testdata/metadata_v1/legacy/connectors-operator/metadata/stage/addon.yaml:44]:
```suggestion
addonOwner: Redacted
```

File: internal/testdata/metadata_v1/legacy/ocm-addon-test-operator/metadata/stage/addon.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  validation: ^test-
+  validation_err_msg: Parameter value must start with test-
+  value_type: string
+addOnRequirements:
+- data:
+    cloud_provider.id: aws
+  enabled: true
+  id: cluster-cloud-provider-req
+  resource: cluster
+addonOwner: Ciaran Roche <croche@redhat.com>
```
Comment (Ajpantuso) [inline internal/testdata/metadata_v1/legacy/ocm-addon-test-operator/metadata/stage/addon.yaml:48]:
```suggestion
addonOwner: Redacted
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Some minor details, but otherwise looks good.

**Comment by Ankit152** [comment]
Done 🙂

**Review by Ajpantuso — APPROVED** [review]
LGTM

**Pull request merged**